### DISCUSSION TITLE
How to update blog to new releases without breaking everything?

### DISCUSSION THREAD

**Discussion by guscastilloa** [alshedivat/al-folio#2320] [Q&A] [answered]
I forked the project 2 years ago and ever since I've seen some amazing updates on the theme. I want to implement some of them but I don't know how the procedure should be. Should I just fork the new one into a new branch and add my _confit, projects, pages, and the corresponding assets? I'm not a web developer so I'm not sure if this would break everything.

**Comment by george-gca** [comment, marked as answer]
Check our [INSTALL.md](https://github.com/alshedivat/al-folio/blob/master/INSTALL.md#upgrading-from-a-previous-version) information.

**Comment by pazz** [comment]
I am in a similar situation (branched off a few years back) but am using a local gitlab to build and publish on our departmental websites, so cannot use the above github action.

One simple way, it seems, to avoid the issue would be to [package al-folio in gem form](https://jekyllrb.com/docs/themes/#creating-a-gem-based-theme).
That way, users like us could simply use the gem, which updates independently, and keep only their personal stuff in the repo.

You could publish this gem via github actions automatically.
This seems such an obvious solution that I suspect there must be a reason why it's not done already?

**Reply by george-gca** [reply]
All the discussion about a gem based theme happened in https://github.com/alshedivat/al-folio/issues/114.

**Comment by pazz** [comment]
@george-gca how do you suggest keeping updated with this theme then? I don't see any other way except manually rebasing every week or so and hoping for the best. 

And yes, i have seen the [related docs](https://github.com/alshedivat/al-folio/blob/master/INSTALL.md#upgrading-from-a-previous-version) but I mean without relying on github pages etc.
Thanks!

**Reply by george-gca** [reply]
Tbh what I do is merge with the tip of the repository every now and then. I don't have a better solution to give you. But at least after we did a lot of cleaning and refactoring in the code, huge changes are unlikely to happen that frequently.

**Reply by pazz** [reply]
Thanks for your response @george-gca.
Conceptually, I still think that maintaining the theme in a gem format and updating it on merges/releases is the cleanest solution.
But of course I do not have the time nor expertise to be more constructive atm, sorry :)

**Reply by george-gca** [reply]
In that matter I agree with https://github.com/alshedivat/al-folio/issues/114#issuecomment-1875676393.
